### DISCUSSION TITLE
Has anyone run into issues with Google Analytics 4 adding `_gl` to URLs that break links to section headers?

### DISCUSSION THREAD

**Discussion by rachel-rigdon-sp** [squidfunk/mkdocs-material#7617] [Q&A] [answered]
We have GA4 in our docs using a script in a custom.html partial override per the [docs](https://squidfunk.github.io/mkdocs-material/setup/setting-up-site-analytics/#custom-site-analytics).

We have a script that pushes links to a Khoros product, but when users click the links there, it appends `_gl` to the links on click and just dumps the user on the base page instead of taking them to the linked section header.

For example, 
> `https://documentation.sailpoint.com/saas/help/workflows/workflow-actions.html#active-directory`

becomes: 
> `https://documentation.sailpoint.com/saas/help/workflows/workflow-actions.html#active-directory&_gl=1*4uv422*_gcl_au*MTYzOTY4MTAwNi4xNzI1OTkwOTg3`

I'm 99% sure this is a conflict with the script on the Khoros system [stopping event propagation before it reaches the document node](https://support.google.com/analytics/answer/10071811?hl=en#troubleshooting) (and I apologize for posting about it here), but I'm doing my due diligence to confirm if others have run into similar issues with a MkDocs-Material and Google Analytics configuration.

**TL;DR: Have y'all run into any issues with GA4 appending gobbledygook that prevents links to section headers?**

**Comment by kamilkrzyskow** [comment]
Hello @rachel-rigdon-sp,
Looking at the example the issue is related to the `&` separator, which shouldn't be there because this attribute is the first to get added, so instead it should be a `?`:
`https://documentation.sailpoint.com/saas/help/workflows/workflow-actions.html#active-directory?_gl=1*4uv422*_gcl_au*MTYzOTY4MTAwNi4xNzI1OTkwOTg3`

**Reply by kamilkrzyskow** [reply]
Not related to the issue, or perhaps not an issue at all, but I opened up the page with `view-source` and there is:
`<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin=>`

`crossorigin` shouldn't have a `=` after it, and I checked that's the case with the `base.html`:
- https://github.com/squidfunk/mkdocs-material/blob/e29e124e426e4ccd16d8937c13e9bdb7ef26d480/src/templates/base.html#L131

So is that `=` added in your custom `base.html` or is it related to the Instant Loading mode 🤔 
Again, it's likely a non-issue, just seems off.

**Reply by rachel-rigdon-sp** [reply]
@kamilkrzyskow Thanks for responding--the separator is helpful info because I'm noticing that if I paste the GA code between the .html and anchor, it takes me to the correct section: 
https://documentation.sailpoint.com/saas/help/workflows/workflow-actions.html?_gl=1*nxvt94*_gcl_au*MTYzOTY4MTAwNi4xNzI1OTkwOTg3#active-directory

Do y'all know if there is a way to change the GA script to insert that code immediately after the `.html` using the correct separator? And if the answer is "that's a question for Khoros and Google" that's totally fine -- I just wanted to check with our esteemed experts here, and the separator info alone has been immensely helpful. 🙌🏻

**Reply by rachel-rigdon-sp** [reply]
> Not related to the issue, or perhaps not an issue at all, but I opened up the page with `view-source` and there is: `<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin=>`
> 
> `crossorigin` shouldn't have a `=` after it, and I checked that's the case with the `base.html`:
> 
> * https://github.com/squidfunk/mkdocs-material/blob/e29e124e426e4ccd16d8937c13e9bdb7ef26d480/src/templates/base.html#L131
> 
![Screenshot 2024-10-14 at 11 47 26 AM](https://github.com/user-attachments/assets/cd5440c2-ee07-46dc-8378-37e9f1a5a659)

> So is that `=` added in your custom `base.html` or is it related to the Instant Loading mode 🤔 Again, it's likely a non-issue, just seems off.

Hmm, I can see that in the comments.html, there is a reference to `crossorigin="anonymous"`. We have a partial override of the comments.html to insert our own feedback system. I'm not sure if that would have an impact. :/ 

I suppose I could try turning instant loading and/or comments off and test links to those pages.

**Reply by kamilkrzyskow** [reply, marked as answer]
The GA script appends the `_gl` and it should be responsible for the proper usage of either `&` or `?`, so I would look for a resolution at the source where this parameter is added.

Maybe some custom script adds/removes parameters to the URL and this happens after the GA script "decides" what separator to use.
So initially there is `URL/?someValue=123`, then GA detects there are params, then your script removes the `?someValue` together with the `?` separator and then GA adds the `&_gl` because it didn't check the URL again before adding the new param 🤔 

So probably "that's a question for Khoros and Google", as it's hard to guess what could be causing the issue.

**Reply by kamilkrzyskow** [reply]
`crossorigin="anonymous"` is fine, I meant the `crossorigin=` without a value, just odd `=` at the end.

**Reply by rachel-rigdon-sp** [reply]
Thank you, as always, for your insight and helping me diagnose this issue, even though I know it isn't a MkDocs-Material thing. 🙏🏻
